packages/astro/src/core/build/static-build.ts

```diff
@@ -177,7 +182,7 @@ async function ssrBuild(opts: StaticBuildOptions, internals: BuildInternals, inp
 		root: viteConfig.root,
 		envPrefix: 'PUBLIC_',
 		server: viteConfig.server,
-		base: astroConfig.buildOptions.site ? new URL(astroConfig.buildOptions.site).pathname : '/',
+		base: astroConfig.buildOptions.site ? fileURLToPath(new URL(astroConfig.buildOptions.site)) : '/',
```

Comment:
just ran into this bug, where this throws when `buildOptions.site` is set (since a value like `'https://docs.astro.build/'` is not a fileURL). Will add a PR to fix